sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/interpreter/BeamSqlFnExecutor.java

```diff
@@ -278,7 +279,9 @@ private static BeamSqlExpression getBeamSqlExpression(RexNode rexNode) {
         case "LIKE":
           ret = new BeamSqlLikeExpression(subExps);
           break;
-
+        case "NOT LIKE":
+          ret = new BeamSqlNotLikeExpression(subExps);
```

Comment:
Can we just pass a flag into `BeamSqlLikeExpression`?

Reply:
Cool, I don't mind it this way.

But I disagree that it is more readable :) It is a whole new separate class for NotLike. If I see it in the code, I would wonder how is it different from Like, and I have to read the whole class now to make sure it works as expected. And in this case I am confused because it defines operation for numbers and dates, so now I also have to read the `Like` class to learn that it's undefined there and throws exceptions. It also adds surface area for errors, now reading this I have to make sure that there's no `!` missing anywhere. Instead I could just read 2 lines of code :) and it doesn't have to be a boolean flag though, e.g.:

```
enum Like { LIKE, NOT_LIKE }
...
like = compare();
return (type == NOT_LIKE) ? !like : like;
```